src/libraries/System.Private.CoreLib/src/System/Text/Encoding.cs

```diff
@@ -1106,7 +1106,7 @@ internal void ThrowCharsOverflow(DecoderNLS? decoder, bool nothingDecoded)
             decoder!.ClearMustFlush();
         }
 
-        internal sealed class DefaultEncoder : Encoder, IObjectReference
+        internal sealed class DefaultEncoder : Encoder
```

Comment:
This is OK to remove because the class is `internal sealed` and therefore never exposed publicly or used internally, right?

Reply:
An instance of this class could be exposed publicly via `Encoding.GetEncoding("some_oddball_encoding").GetEncoder()`. The caller would then have to cast this object to an `IObjectReference`. With this change, that cast would now fail. However, I can't think of a case where anybody would ever run into this situation, since: (a) it relies on somebody having subclassed `Encoding`; (b) and that person not having overridden these APIs; (c) and _somebody else_ making an explicit cast of that nonpublic object to this interface. I have high confidence that the intersection of all of these is exactly zero people.

Regardless, this is the type of breaking change we would take. There's no guarantee from major version to major version that we'll ever return the same concrete type of the internal object. We even changed `Encoding.Default` in past releases.